core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java

```diff
@@ -121,7 +121,7 @@ public ByteBuffer readRemoteBlock(InetSocketAddress address, long blockId, long
       }
       throw new IOException(e);
     } finally {
-      if (channel != null && listener != null) {
+      if (channel != null && listener != null && channel.isActive()) {
```

Comment:
nit: do you also want to check `channel.pipeline().get(ClientHandler.class)` is not-null ? Is it possible in some corner cases, when the channel is still active, but `ClientHandler` is somehow removed from the pipleline?